### ISSUE TITLE
Double scan on iOS

### ISSUE THREAD

**Issue by DottorPagliaccius** [JimmyPun610/BarcodeScanner.XF#12]
Hi, great job.

CameraView_OnDetected is fired two times (sometimes even more than two) on iOS (android works fine).

You can reproduce it in the Sample App.

If I may, I'll suggest to fire CameraView_OnDetected in the Main Thread.

Thank you,
Dottor Pagliaccius

**Comment by JimmyPun610** [comment]
@DottorPagliaccius 
Thank you. I moved the event action part to UIThread now.
```C#
  InvokeOnMainThread(() => {
                    CaptureSession.StopRunning();
                    this.OnDetected?.Invoke(list); 
                });
```
Previously
```C#
this.OnDetected?.Invoke(list); 
CaptureSession.StopRunning();
```
For the bugs, I made below change and tested the situation is not occurring in my iPhone 11.
I am not sure about the case is solved not or.

**Comment by DottorPagliaccius** [comment]
My device is an iPhone 7 with iOS 13.3, sadly I've no chance to test it with other device at the moment. I will try the new version and let you know.

I think it should fired in the Main Thread in Android too. 

Thank you.

**Comment by DottorPagliaccius** [comment]
nope, just tried the new version, in the sample app scan fires multiple events :/

**Comment by JimmyPun610** [comment]
@DottorPagliaccius 
Updated the onDetected event will run on UIThread for both platform now.
Do you have any hints about what is the step of firing multiple event?
The situation is not happening on my side and I cannot debug it

**Comment by JimmyPun610** [comment]
Finally I note that it is because DidOutputSampleBuffer is calling very often.
Hopefully I can try to solve it

**Comment by JimmyPun610** [comment]
Added 1 second interval on DidOutputSampleBuffer 
Hopefully it solves the bug.
Available on 4.2.1.2
